core/primitives/src/crypto.rs

```diff
@@ -510,11 +511,22 @@ mod dummy {
 		fn as_ref(&self) -> &[u8] { &b""[..] }
 	}
 
+	#[allow(clippy::transmute_ptr_to_ptr)]
 	impl AsMut<[u8]> for Dummy {
 		fn as_mut(&mut self) -> &mut[u8] {
 			unsafe {
 				#[allow(mutable_transmutes)]
+				// Suggested #2
+				//let uc = &UnsafeCell::new(&b""[..]);
+
+				// Original
 				rstd::mem::transmute::<_, &'static mut [u8]>(&b""[..])
+
+				// Suggested #1
+				// &mut *(&b""[..] as *const [u8] as *mut [u8])
+
+                // Suggested #2
+				//&mut *uc.get()
```

Comment:
I don't know what to prefer. maybe suggest 1

Reply:
I forgot to remove that comment. This was a `Dummy` datatype, it's not worth it to implement these. I tried it, and the borrow checker does not make it easy.